docs/src/literate/helmholtz.jl

```diff
@@ -0,0 +1,169 @@
+    
+    n_basefuncs = getnbasefunctions(cellvalues)
+    global_dofs = zeros(Int, ndofs_per_cell(dh))
+
+    fe = zeros(n_basefuncs) # Local force vector
+    Ke = zeros(n_basefuncs, n_basefuncs) # Local stiffness mastrix
+
+    @inbounds for (cellcount, cell) in enumerate(CellIterator(dh))
+        fill!(Ke, 0)
+        fill!(fe, 0)
+        coords = getcoordinates(cell)
+
+        reinit!(cellvalues, cell)
+        # First we derive the non boundary part of the variation problem from the destined solution `u_ana`
+        # ```math
+        # \int \nabla δu \cdot \nabla u d\Omega
+        # + \int δu \cdot u d\Omega
+        # - \int δu \cdot f d\Omega
+        # ```
+        #+
+
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+            coords_qp = spatial_coordinate(cellvalues, q_point, coords)
+            f_true = LinearAlgebra.tr(hessian(u_ana, coords_qp)) + u_ana(coords_qp)
+            for i in 1:n_basefuncs
+                δu = shape_value(cellvalues, q_point, i)
+                ∇δu = shape_gradient(cellvalues, q_point, i)
+                fe[i] += (δu * f_true) * dΩ
+                for j in 1:n_basefuncs
+                    u = shape_value(cellvalues, q_point, j)
+                    ∇u = shape_gradient(cellvalues, q_point, j)
+                    Ke[i, j] += (∇δu ⋅ ∇u + δu * u) * dΩ
+                end
+            end
+        end
+
+        # Now we manually add the von Neumann boundary terms
+        # ```math
+        # \int δu \cdot (n \cdot \nabla u - g_2) d\Gamma_2
+        # ```
+        #+
+        for face in 1:nfaces(cell)
+            if onboundary(cell, face) && 
+                   ((cellcount, face) ∈ getfaceset(grid, "left") || 
+                    (cellcount, face) ∈ getfaceset(grid, "bottom"))
+                reinit!(facevalues, cell, face)
+                for q_point in 1:getnquadpoints(facevalues)
+                    coords_qp = spatial_coordinate(facevalues, q_point, coords)
+                    n = getnormal(facevalues, q_point)
+                    g = gradient(u_ana, coords_qp) ⋅ n
+                    dΓ = getdetJdV(facevalues, q_point)
+                    for i in 1:n_basefuncs
+                        δu = shape_value(facevalues, q_point, i)
+                        fe[i] += (δu * g) * dΓ
+                    end
+                end
+            end
+        end
+   
+        celldofs!(global_dofs, cell)
+        assemble!(assembler, global_dofs, fe, Ke)
+    end
+    return K, f
+end;
+
+K, f = doassemble(cellvalues, facevalues, K, dh);
+apply!(K, f, dbcs)
+u = Symmetric(K) \ f;
+
+vtkfile = vtk_grid("helmholtz", dh)
+vtk_point_data(vtkfile, dh, u)
+vtk_save(vtkfile)
+
+using Test
+# TODO where does this number come from?
+Test.@test maximum(u) ≈ 0.05637592090022005
+# TODO would be nice to actually compute e.g. the L2 error between the analytic solution
+# (projected to the fem space) and the numeric solution.
```

Comment:
Some guidance would be appreciated. I have no idea, where this number comes from. 
Also, I would really like to compute some error norms. But it seems, that `JuAFEM` does not really have much infrastructure to do that (no struct that represents an element of a FEM space, no way of projecting a `Function` onto a FEM space). Is that correct?

Reply:
> Some guidance would be appreciated. I have no idea, where this number comes from.

I think that number is just the value we got, not compared to any other software. The purpose is just to make sure it does not regress.

> I would really like to compute some error norms. But it seems, that JuAFEM does not really have much infrastructure to do that (no struct that represents members of a FEM space, no way of projecting a Function onto a FEM space). Is that correct?

Yea, you have to roll your own function for that, but I think it should not be too difficult. So far the functionality in here is mostly what we have needed ourselves for our own research projects, but what you are talking about would be a great addition.

Reply:
It is just the number that I got... not scientific but better than nothing 🤷‍♂

Reply:
I might add some helper functions to compute error norms in the future. But not in this PR.